core/src/avm2/globals/flash/geom/point.rs

```diff
@@ -0,0 +1,416 @@
+//! `flash.geom.Point` builtin/prototype
+
+use crate::avm1::AvmString;
+use crate::avm2::class::Class;
+use crate::avm2::method::Method;
+use crate::avm2::traits::Trait;
+use crate::avm2::{Activation, Error, Namespace, Object, QName, TObject, Value};
+use gc_arena::{GcCell, MutationContext};
+
+fn create_point<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    coords: (f64, f64),
+) -> Result<Value<'gc>, Error> {
+    let proto = activation.context.avm2.prototypes().point;
+    let args = [Value::Number(coords.0), Value::Number(coords.1)];
+    let new_point = proto.construct(activation, &args)?;
+    instance_init(activation, Some(new_point), &args)?;
+
+    Ok(new_point.into())
+}
+
+/// Implements `flash.geom.Point`'s instance constructor.
+pub fn instance_init<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    let _ = set_to(activation, this, args)?;
+    Ok(Value::Undefined)
+}
+
+fn coords<'gc>(
+    this: &mut Object<'gc>,
+    activation: &mut Activation<'_, 'gc, '_>,
+) -> Result<(f64, f64), Error> {
+    let x = this
+        .get_property(*this, &QName::new(Namespace::public(), "x"), activation)?
+        .coerce_to_number(activation)?;
+    let y = this
+        .get_property(*this, &QName::new(Namespace::public(), "y"), activation)?
+        .coerce_to_number(activation)?;
+    Ok((x, y))
+}
+
+fn set_coords<'gc>(
+    this: &mut Object<'gc>,
+    activation: &mut Activation<'_, 'gc, '_>,
+    value: (f64, f64),
+) -> Result<(), Error> {
+    this.set_property(
+        *this,
+        &QName::new(Namespace::public(), "x"),
+        value.0.into(),
+        activation,
+    )?;
+    this.set_property(
+        *this,
+        &QName::new(Namespace::public(), "y"),
+        value.1.into(),
+        activation,
+    )?;
+    Ok(())
+}
+
+/// Implements `flash.geom.Point`'s class initializer.
+pub fn class_init<'gc>(
+    _activation: &mut Activation<'_, 'gc, '_>,
+    _this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    Ok(Value::Undefined)
+}
+
+/// Implements the `length` property
+pub fn length<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(mut this) = this {
+        let (x, y) = coords(&mut this, activation)?;
+
+        return Ok((x.powf(2.0) + y.powf(2.0)).sqrt().into());
```

Comment:
I'd prefer `x*x + y*y` here (even though rustc is smart enough to optimize this).